R/utils.R

```diff
@@ -60,18 +60,26 @@ setMethod("variantFilter",
 
 # index is in case we had to subset geno so it no longer matches the variant filter
 # (in the case of allele matching)
-.alleleFreq <- function(gdsobj, geno, variant.index=NULL, sample.index=NULL, male.diploid=TRUE, genome.build=c("hg19", "hg38")) {
+.alleleFreq <- function(gdsobj, geno, variant.index=NULL, sample.index=NULL, male.diploid=TRUE, genome.build=c("hg19", "hg38"),
+                        geno.test=c('additive','recessive','dominant')) {
 
     # check sex
     sex <- validateSex(gdsobj)
     if (!is.null(sample.index)) sex <- sex[sample.index]
     if (is.null(sex)) {
         #freq <- 0.5*colMeans(geno, na.rm=TRUE)
         count <- colSums(geno, na.rm=TRUE)
-        # nsamp <- colSums(!is.na(geno))
+        ## nsamp <- colSums(!is.na(geno))
         nsamp <- .countNonMissing(geno, MARGIN = 2)
-        freq <- count/(2*nsamp)
-        mac <- round(pmin(count, 2*nsamp - count))
+        
+        if(genoTest=='additive'){
+            freq <- count/(2*nsamp)
+            mac <- round(pmin(count, 2*nsamp - count))
+        }else{ # recessive or dominant
+            freq <- count/nsamp
+            mac <- count
+        }
+        
```

Comment:
For recessive, `freq` is the frequency of homozygous alts (i.e. the proportion of 1s) and `MAC` is the count of hom alts. 
For dominant, `freq` is the frequency of hets or hom alts (i.e. the proportions of 1s) and `MAC` is the count of hets and hom alts. 

@smgogarten and @amstilp - does this make sense to you? I think `freq` makes sense to me, but I'm less sure about `MAC`. Should `MAC` be the count of the smaller group? maybe that's more consistent with "minor". We're clearly abusing the name `MAC` here, but I don't think we want to deal with different names of the variable depending on the genotype coding, right?

For recessive, `MAC` is

Reply:
I don't think we should change the column name, and I think `MAC` should always count the smaller group. That would be `mac <- round(pmin(count, nsamp - count))`, right?

In the man page, we should include a section in `Details` listing things that are different for recessive and dominant coding, including how we defined `freq` and `MAC`.